documentation/APIs/core-models.md

```diff
@@ -0,0 +1,581 @@
+            <td>
+              The type of the payload. It will instruct the node how to parse the fields that follow.
+            </td>
+          </tr>
+          <tr>
+            <td>Payload Bytes</td>
+            <td>ByteArray</td>
+            <td>A sequence of serialized bytes, where the structure depends on `Payload Type`.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td valign="top">Signature <code>oneOf</code></td>
+    <td colspan="2">
+      <details open="true">
+        <summary>Generic Signature</summary>
+        <table>
+          <tr>
+            <th>Name</th>
+            <th>Type</th>
+            <th>Description</th>
+          </tr>
+          <tr>
+            <td>Signature Type</td>
+            <td>uint8</td>
+            <td>
+              The type of the signature. It will instruct the node how to parse the fields that follow.
+            </td>
+          </tr>
+          <tr>
+            <td>Signature Bytes</td>
+            <td>ByteArray</td>
+            <td>A sequence of serialized bytes, where the structure depends on `Signature Type`.</td>
+          </tr>
+        </table>
+      </details>
+    </td>
+  </tr>
+  <tr>
+    <td>Nonce</td>
+    <td>uint64</td>
+    <td>The nonce which lets this block fulfill the PoW requirement.</td>
+  </tr>
+</table>
+
+
+### Block ID
+Block ID denotes an identifier of a block, with type `ArrayBytes[40]`. It is calculated as the following steps, using BLAKE2b-256 hash function:
+
+* `content` is the serialized block **without** signature and nonce.
+* `slot_index` is the slot index of the `Issuing Time` of the block.
+* `signatureBytes` is the serialized Ed25519 signature.
+
+Calculation:
+1. `content_hash` = hash(`content`)
+2. `id` = hash(Concat(`content_hash`, signatureBytes))
+3. And finally, `BlockID` = Concat(`id`, `slot_index`) 
+
+The string format of Block ID is hexadecimal encoding of Block ID with `0x` prefix.
+
+### Block Syntactic Validation Rules
+
+* Block size must not exceeds `MaxBlockSize` bytes, currently is `32768`.
+* `Protocol Version` must match the `Protocol Version` config parameter of the node.
+* Must have at least **1** `Strong Parents`.
+* `Strong Parents`, `Weak Parents`, `Shallow Like Parents` must not exceeds `BlockMaxParents` (currently is **8**) per type.
+* `Strong Parents`, `Weak Parents`, `Shallow Like Parents` of parents must be lexically ordered.
+* `Strong Parents`, `Weak Parents`, `Shallow Like Parents` must not have duplicates in each list.
+* `Weak Parents` must be disjunct to the rest of the parents.
+* `Payload Type` must match one of the values described under Payloads.
+  Data Fields must be correctly parsable in the context of the `Payload Type`.
+  The payload itself must pass syntactic validation.
+* `Signature` must be valid, the input of a signature is concatation of Issuing Time bytes, Commitment ID bytes and Block bytes (without Signature and Nonce) by order.
```

Comment:
Isn't `Issuing Time` included twice in the signature then ?

Reply:
Same as Commitment ID?

Reply:
Firstly, Block bytes are not the input of the Signature, it's `content_hash` instead, already fixed it in the doc. :sweat_smile:
 
And yes, we'll need the Issuing Time explicitly to perform validation in slot commitment chain fork.

Reply:
What do you mean it's been fixed? I still see this:
> Signature must be valid, the input of a signature is concatation of Issuing Time bytes, Commitment ID bytes and Block bytes (without Signature and Nonce) by order.

Reply:
Where is `Slot Commitment ID` from? I assume that's *not* the one from within the Block itself?

Reply:
It's the ID of Slot Commitment in a Block. I'll add this description.

Reply:
Ok so it's `Previous Slot Commitment ID`

Reply:
Nope, it's the ID calculated from the Slot Commitment structure of the block.

Quote from the Slot Commitment ID section:

Slot Commitment ID is calculated as the following steps, using BLAKE2b-256 hash function:

* `content` is the serialized slot commitment.
* `slot_index` is the slot index of the slot commitment (`Index` from the Slot Commitment Structure).

Caculation:
1. `content_hash` = hash(`content`)
2. And finally, `CommitmentID` = Concat(`content_hash`, `slot_index`)

Reply:
Let me add a Block Signature section to go through the details.